chem_evol.py

```diff
@@ -3469,6 +3526,86 @@ def func_total_ejecta(inimass):
+				self.stellar_param.append([0.]*self.nb_timesteps)
+				#linear spline fit between grid points
+				attr_data=[]
+				for h in range(len(m_stars_grid)):
+					attr_data.append(self.ytables.get(M=m_stars_grid[h],Z=self.Z_gridpoint,quantity=self.ytables.col_attrs[k]))			
+				f = interp1d(m_stars_grid, attr_data)
+				self.stellar_param_functions.append(f)	
+		#tabulated parameter	
+		table_p=self.table_param #to be better readable
+		stellar_param_evol_num=len(table_p.data_cols)-1 #-age
+		for h in range(1,len(table_p.data_cols)):
+			self.stellar_param_attrs.append(table_p.data_cols[h])
+			self.stellar_param.append([0.]*self.nb_timesteps)
+	
+		stellar_param_evol_num=len(table_p.data_cols)-1 #-age
+		dts= self.history.timesteps		
+		stellar_param_evol=[]
+		#loop over all stars
+		for h in range(len(m_stars_grid)):
+        		ages=table_p.get(Z=self.Z_gridpoint,M=m_stars_grid[h],quantity='Age')
+			#get the parameter from tables
+			stellar_param_table_evol=[]
+			stellar_param_evol1=[]
+			for k in range(1,len(table_p.data_cols)):
+				stellar_param_table_evol.append(table_p.get(Z=self.Z_gridpoint,M=m_stars_grid[h],quantity=table_p.data_cols[k]))
+				stellar_param_evol1.append([0.]*len(dts))
+
+			#fill the time step bins with values
+			age_sim_last=0.
+			age_sim=0.
+			ages=[0.]+list(ages)
+			for k in range(len(dts)):
+				# for timestep loop over table time entries, are they relevant for time step bin? 
+				# during which fraction frac of the time step bin are they relevant?
+				# create mean value for each time step bin by taking account the fractions
+				age_sim = age_sim + dts[k]
+                	        for t in range(1,len(ages)):
+					#to prevent same time entries to add up; stop then
+					if ages[t] == ages[t-1]:
+						break
+					#table grid step: ages[t-1] - ages[t]
+                        	        if ages[t]<age_sim_last:
+						continue
+					elif ages[t-1]>age_sim:
+						continue
+					elif ages[t-1]<=age_sim_last and ages[t]<=age_sim:
+						frac = (ages[t] - age_sim_last)/dts[k]
+						for p in range(stellar_param_evol_num):
+							stellar_param_evol1[p][k] += stellar_param_table_evol[p][t-1] * frac		
+					elif ages[t-1]>=age_sim_last and ages[t]>age_sim:
+						frac = (age_sim - ages[t-1])/dts[k]	
+	                	        	for p in range(stellar_param_evol_num):
+                        	                        stellar_param_evol1[p][k] += stellar_param_table_evol[p][t-1] * frac
+					else:
+						frac = (ages[t] - ages[t-1])/dts[k]
+                        	                for p in range(stellar_param_evol_num):
+                        	                        stellar_param_evol1[p][k] += stellar_param_table_evol[p][t-1] * frac
+					if frac<0:
+						print 'error frac!!!, t: ',t
+				age_sim_last=age_sim
+			stellar_param_evol.append(stellar_param_evol1)
+	
+
+		self.stellar_param_evol_masses=m_stars_grid
+		self.stellar_param_evol=stellar_param_evol		
+
```

Comment:
This is a solid segment of code!  Is this inside an already-existing function?  Is so, I would suggest that you put the new segment in a separate function in order to avoid having too long functions.  But this will not hold me back from merging the branch ;)